pkg/rancher-components/src/components/BadgeState/BadgeState.vue

```diff
@@ -0,0 +1,100 @@
+<script>
```

Comment:
If this does not slow you down, may we use Typescript for these first isolated components?

Reply:
Ok forget it, I would rather avoid this till we setup correctly the project to support Typescript. I just tried to create a test for a TS component and this had some issues :cry:

Reply:
I think this is a good idea and might be the next thing we could tackle. I don't anticipate there being any issue, but I'd like to keep the scope of this PR focused on standing up the component library and follow up with some long overdue refactoring in subsequent PRs.

Reply:
Created #5932 so we can address this

Reply:
@rak-phillip in the meanwhile the issue with TS has also been solved, but we can create tests afterwards without problems. I've created a bunch of tests cases if you are interested to take a look into: https://github.com/rancher/dashboard/pull/5779